### PR TITLE
RCG-30: Description deduplication

### PR DESCRIPTION
**Pull request by tsellers-r7**
## Description
This PR:

- Reduces the number of duplicate `description` fields. Duplicates in this field can cause issues when processing large number of banners and generating metrics on the fingerprint matches.
- Cleans up a few multi-line `description` fields, these are undesirable for similar metrics reasons.
- Adds a few `service.version` tests for `example` fields. Every regex should have every extracted value tested at least once. One of these new tests found an error in the regex for a fingerprint.

Expect a few more of these as time permits. I'm trying to avoid asking folks to review a wall of changes all at once.

## Motivation and Context
These changes improve the usability of `recog`

## How Has This Been Tested?
`rspec`

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jhart-r7** [comment]
Does this fix all duplication?  If so, we should put a test in place to enforce it.

**Comment by tsellers-r7** [comment]
@jhart-r7  - No. Expect a few more of these as time permits. I'm trying to avoid asking folks to review a wall of changes all at once.

File: xml/x509_subjects.xml [hunk 1]
```diff
@@ -878,7 +878,7 @@
     <param pos="0" name="os.device" value="WAP"/>
   </fingerprint>
   <fingerprint pattern="^CN=axis-([a-fA-F0-9]{12}),O=Axis Communications AB$">
-    <description>OpenWRT WAP</description>
+    <description>OpenWRT WAP - Axis Communications variant</description>
```
Comment (jhart-r7) [inline xml/x509_subjects.xml]: I suspect this was a copy-past problem from the fingerprint above.  I don't believe this is OpenWRT.

Comment (tsellers-r7) [reply xml/x509_subjects.xml]: Thanks, I should have caught that. I caught similar in one or two other places.

**Comment by hdm** [comment]
Thanks for catching it!

**Pull request merged**